equibop: 2.1.7 -> 3.1.7

Rexcrazy804 opened the pull request:
- swap out pnpm with bun following upstream changes for 3.X.X
- added myself as maintainer
- removed outdated patch
- added `bun` to wrapper's `PATH`

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [x] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Rexcrazy804 committed:
equibop: 2.1.7 -> 3.1.4

justDeeevin reviewed (approved):
great work!

NotAShelf reviewed (dismissed):
Upstream has advised us to wait until the next stable release, where bun has been removed entirely. Let's wait until v3.1.0 or v3.0.10 for the merge. Would save us from a second rewrite of the package, and hopefully some of the FOD tomfoolery.

Relevant: https://canary.discord.com/channels/1173279886065029291/1261553921378484286/1433709707771514971

Rexcrazy804 commented:
> Upstream has advised us to wait until the next stable release, where bun has been removed entirely.

Ah, they are jumping ship from bun? That's prolly for the better. I'll keep an eye on the releases.

Rexcrazy804 commented:
bumped to 3.1.0 looks like the arRPC no longer depends on bun, and that's about it.
Additionally, the update script is broken for the node-modules FOD, I'll figure it out as early as I can.

Rexcrazy804 commented:
>  I had to override its outputHash to sha256-EHufreIumqHAEf/SB85+XOGd9Ef5025o31jRH5VEZO0= in order for it to build

This is concerning, I am presently getting `sha256-0wANnIQYFPVIJG15NKJBPzvtoqZ+3ckFDAWAzy+dCBk=`.
I suspect the module's drv isn't reproducible :/

justDeeevin commented:
yes the new hash works and arRPCBun is working :D

mdaniels5757 commented:
Re-running CI due to a since-fixed unrelated treefmt-nix failure

basn commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 456790`
Commit: `bdbc4678510e86f8e2f00005f15633b953902463`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>equibop</li>
  </ul>
</details>

mochienya commented:
I get the same error (with the same hashes) using `nixpkgs-review`, but running
```bash
nix build github:Rexcrazy804/nixpkgs/update-equibop#equibop
```
builds fine. I've [been using equibop on my system](https://github.com/mochienya/nixConfig/blob/5e2c7e91994b9474f7075930f4ed0b5f4ee7d3a4/home.nix#L29-L32) from the fork for the past few days and haven't noticed any issues.

Rexcrazy804 commented:
I've rebased on top of the lastest nixos-unstable commit and this resulted in a new hash for the FOD :/
I'll be taking a closer look at what's changed and how I can avoid this

Rexcrazy804 commented:
There is some randomness causing the hash for the bunDeps(node-modules) to change, (one binary file called eslint) when something changes in nixpkgs. Not much besides that

pyrox0 committed:
equibop: 3.1.4 -> 3.1.6

electron-dist changes copied from
https://github.com/NixOS/nixpkgs/pull/476347

Icon changes from https://github.com/NixOS/nixpkgs/issues/453095

Rexcrazy804 committed:
equibop: 3.1.6 -> 3.1.7
